tests/platform_tests/test_link_down.py

```diff
@@ -158,20 +164,33 @@ def test_link_down_on_sup_reboot(duthosts, localhost, enum_supervisor_dut_hostna
     # RP doesn't have any interfaces, check all LCs' interfaces
     link_status_on_all_LC(duthosts, localhost, fanouts_and_ports, up=False)
 
-    time.sleep(MAX_TIME_TO_REBOOT)
+    # Wait for ssh port to open up on the DUT
+    wait_for_startup(duthost, localhost, 0, MAX_TIME_TO_REBOOT)
```

Comment:
can you move `wait_for_startup` before `link_status_on_all_LC`?

Reply:
The intend here is that when the LC's are in the process of rebooting (not up yet), the ports on remote end (fanouts) are all down. It could happen that by the time that ssh port on supervisor is open,  LCs (that reboot before the supervisor) are back.

Reply:
I mean, after reboot, we need to make sure sup is back up with `wait_for_startup` before checking links status on linecards with `link_status_on_all_LC`. If sup is down, linecards won't be up

Reply:
link_status_on_all_LC is calling link_status_on_host, which doesn't query/send any command to the duthost (LC). So, don't need to check for that. Also, in _link_status_on_host we are polling using wait_until - that would cover the scenario where mgmt connectivity to LC is not there.

Reply:
no,  link_status_on_all_LC checks links status on every LC
`link_status_on_host(LC, localhost, fanouts_and_ports, up)`

Reply:
`link_status_on_host we are polling using wait_until - that would cover the scenario where mgmt connectivity to LC is not there.`  -- I agree, so it also made sure sup is back up, then no need for `wait_for_startup` maybe

if sup is down, linecards won't be up

Reply:
link_status_on_host is polling on links_up method, which takes only fanout, and ports as argument. So, duthost in link_status_on_host is not used at all.

Reply:
oh right.. then it's good to add it here, thanks for catching this

Reply:
I'll raise a PR after yours to remove this `duthosts` argument